src/System.Data.Odbc/src/System/Data/Odbc/OdbcConnection.cs

```diff
@@ -561,7 +562,14 @@ internal void HandleError(OdbcHandle hrHandle, ODBC32.RetCode retcode)
 
         public override void Open()
         {
-            InnerConnection.OpenConnection(this, ConnectionFactory);
+            try
+            {
+                InnerConnection.OpenConnection(this, ConnectionFactory);
+            }
+            catch (DllNotFoundException e) when (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                throw new OdbcException(SR.Odbc_UnixOdbcNotFound + Environment.NewLine + e.Message, new OdbcErrorCollection());
```

Comment:
This would look like this

```
System.Data.Odbc.OdbcException : Could not find dependency. Please check that unixOdbc is installed.
        Unable to load DLL 'odbc.so.2': The specified module or one of its dependencies could not be found.
         (Exception from HRESULT: 0x8007007E)
        Stack Trace:
           /home/maryam/git/corefx/src/System.Data.Odbc/src/System/Data/Odbc/OdbcConnection.cs(571,0): at System.Data.Odbc.OdbcConnection.Open()
           /home/maryam/git/corefx/src/System.Data.Odbc/tests/DependencyCheckTest.cs(17,0): at System.Data.Odbc.Tests.DependencyCheckTest.OdbcConnection_OpenWhenOdbcNotInstalled_ThrowsOdbcException()
```

Reply:
Can you make it a DllNotFoundException? The only purpose of this change is to make the message of the DLLNFE more helpful, there's no need to change hte type.